### PR TITLE
perf(PWA): Use Network-only strategy for service worker

### LINKED ISSUE #1350: Strange behavior of login with authentication bypass
I use the hotio docker of qbt with the vue theme. It is behind traefik, protected by the Authelia SSO portal.
In settings/webui I selected "Bypass authentication for clients whitelisted IP subnets" and put in all private networks:
- 10.0.0.0/8
- 172.16.0.0/12
- 192.168.0.0/16

The goal was that qbt would not ask for a password at all, since the traffic always comes from a private network via traefik. And the UI is protected by Authelia.

This led to a very inconsistent experience.

- The first login on a fresh browser (nothing in cache/ no cookies) works always fine
- Every second visit of the page has mixed outcomes:
  -  Almost always I can see the sidebar (with actual stats). Sometimes with torrents in the main window, sometimes with the login mask. The menu on top is also visible. I can click on "log" and close it again and I see my torrent list. Putting in my credentials to the login mask usually leads to an error. Refreshing the page sometimes also removes the login and shows the torrents.

### PR DESCRIPTION
**Pull request by Larsluph**
Fixes #1350

### CODE REVIEW

**Comment by WDaan** [comment]
🤔

**Comment by WDaan** [comment]
What's the impact of this?

**Comment by Larsluph** [comment]
There shouldn't be any impact, I built the project and copied the generated manifest and service worker from the Vite PWA plugin. And modified the service worker to disable caching.

After the investigation in the linked issue, caching the UI is causing some problems when using a reverse proxy and custom SSO auth.

**Review by WDaan — APPROVED** [review]

**Comment by WDaan** [comment]
I do seem to notice a couple changes since this change.

But would need to deep dive if this is really the cause and how this change could have impacted it.

File: public/registerSW.js
Comment (ICE-GB) [inline public/registerSW.js:1]: Should we use `. /`? I'm using the nginx proxy with path `a.b.c/qb/`

File: public/registerSW.js [hunk 1]
```diff
@@ -0,0 +1 @@
+"serviceWorker"in navigator&&window.addEventListener("load",(async()=>{await navigator.serviceWorker.register("/sw.js")}));
```
Comment (ICE-GB) [inline public/registerSW.js:1]: Should we use `. /` ? I'm using the nginx proxy with path `a.b.c/qb/`

Comment (Larsluph) [reply public/registerSW.js:1]: I just created a PR to address that issue:

https://github.com/VueTorrent/VueTorrent/pull/1511

**Pull request merged**